GpsDataCapturer/wear/src/main/java/com/google/gpsdatacapturer/GpsMainActivity.java

```diff
@@ -0,0 +1,235 @@
+package com.google.gpsdatacapturer;
+
+import android.Manifest;
+import android.app.Dialog;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.content.pm.PackageManager;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.os.IBinder;
+import android.provider.Settings;
+import android.support.wearable.activity.WearableActivity;
+import android.support.wearable.view.WearableDialogHelper;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+import androidx.core.app.ActivityCompat;
+
+import com.google.sharedlibrary.GpsDataCaptureService;
+import com.google.sharedlibrary.GpsDataCaptureService.GpsDataCaptureBinder;
+
+public class GpsMainActivity extends WearableActivity {
+    private static final String TAG = "GpsMainActivity";
+    private static GpsDataCaptureService gpsDataCaptureService;
+    private static Intent serviceIntent;
+    private LocationManager locationManager;
+    private static boolean isBound = false;
+    private static final int LOCATION_REQUEST_CODE = 1;
+    private static final int READ_WRITE_REQUEST_CODE = 2;
+    private static boolean isStart = true;
+    private boolean isGpsEnabled = false;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_gps_main);
+
+        locationManager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
+
+        requestLocationPermissionsIfNotGranted();
+        requestReadWritePermissionsIfNotGranted();
+
+        checkGpsStatus();
+        setGpsEnabled();
+
+        startAndBindGpsDataCaptureService();
+
+        final Button startAndStopButton = (Button) findViewById(R.id.start_stop_button);
+        startAndStopButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (isStart) {
```

Comment:
"isStart" may be confusing name.
if isCaptureStarted - would need reversed logic here.

or it could be enum ButtonState { START_CAPTURE, STOP_CAPTURE } 
then decision here will look like:
if ( startStopButtonState == ButtonState.START_CAPTURE) {
...